### PR TITLE
Feature/whitelist

### PR DESCRIPTION
**Pull request by Johanndutoit**
Creates a section for the whitelist function.

That will allow the hub to define a list of whitelist per group that the users are allowed to access.

The new captive will then pick this list up and allow the users to access the urls by creating a whitelist on the Mikrotik every 2 hours.

### CODE REVIEW

**Comment by Johanndutoit** [comment]
Seems there was a built failure, having a look

**Comment by Johanndutoit** [comment]
Just got the coverage up to par to cover the entire new code

**Comment by smn** [comment]
How does this end up configuring the mikrotik?

**Comment by Johanndutoit** [comment]
Only updates the database that includes a formatted list to the Node. The Captive watches and updates the whitelist every 2 hours if something has changed along with the app update check.

**Comment by smn** [comment]
I'm probably missing but how is that white list written & relayed to the nodes?

File: goddard/handlers/handshake/setup.coffee [hunk 1]
```diff
@@ -28,14 +26,50 @@ module.exports = exports = (app) ->
...
+							where: {
+
+								'groupId': node_obj.groupId
+
+							}
+
+						}).then (whitelist_objs) ->
+
+							# append the whitelist
+							public_response_obj.whitelist = []
```
Comment (Johanndutoit) [inline goddard/handlers/handshake/setup.coffee:47]: @smn whitelist is set here at /setup.json which is where the nodes handshake

Comment (smn) [inline goddard/handlers/handshake/setup.coffee:47]: thanks!

**Comment by Johanndutoit** [comment]
@jonathanendersby @smn to approve

**Comment by smn** [comment]
:+1: from me.

**Pull request merged**